certbot-dns-rfc2136/certbot_dns_rfc2136/dns_rfc2136.py

```diff
@@ -167,6 +173,30 @@ def del_txt_record(self, domain_name, record_name, record_content):
             raise errors.PluginError('Received response from server: {0}'
                                      .format(dns.rcode.to_text(rcode)))
 
+    def _get_ip_for_hostname(self, hostname):
```

Comment:
Is there a way to write some meaningful unit tests for this method? (Perhaps one where `hostname` is an IP address, one where `hostname` is expected to resolve successfully, and one where `hostname` is expected to fail to resolve?)

Among other benefits, it would be nice for Travis to continually verify that the `socket.getaddrinfo` invocation below works on all supported platforms and Python versions.

Reply:
Yes, I think there is at least some minimal testing we can do here. I have written some basic tests that do not depend on external services:

https://github.com/eriktews/certbot/commit/c26cee88e07adb9579b3a61a94279004a75f493d

Also the spelling has been corrected.

My branch has bee updated with those improvements.

Reply:
Unfortunately Travis CI does not yet support IPv6, so the IPv6 test has to be disabled for now. It passes fine on my local system.